### DISCUSSION TITLE
Big improvements: How to make the app unique

### DISCUSSION THREAD

**Discussion by Benji377** [Benji377/OpenMusic#122] [Ideas]
There has been an idea on my mind for quite a while now. I wanted to make the app unique in its genre. There are a lot of other music player apps out there which can do just the same as our app does. Therefore here comes my crazy stupid solution:
Add the ability to share music with other devices with the same app on it.
There are two ways I figured out:
1. Create like a messaging app that instead of sending messages just sends mp3 files
2. Exchange files with other devices in WiFi or Bluetooth range.

Number two is definetily the easier solution and I think its the solution we should go for if we decide to implement this. Number one would need to implement a list of contacts or something and that would be a lot of trouble, furthermore it would make the app no longer a 100% offline app which I am kind of aiming for.

I knows its a big idea, but let me know what you think about it.

**Comment by squivix** [comment]
Getting really ambitious aren't you, Benji? :P

I'd say this is waaay low on the priority list. Right now I'm working on issue #125. Then I'm thinking I'd help with the playlist thing. Next, we'll work on adding song metadata and expand the app out gradually to have all the features that a user expects from a media player app. THEN we should think about features the user wouldn't expect like this one.

**Comment by Benji377** [comment]
Ok yeah, you are right hahahha. I may have exaggerated on this one.
By the way, the Playlist should be alright now, all left to do is to create a Fragment that manages the playlists. Where the user can create playlists, delete them, add and or remove songs from them.
The main functions should be working and arent too hard to understand.
